tests/test_locality_NeighborList.py

```diff
@@ -180,6 +180,62 @@ def test_from_arrays(self):
             nlist = freud.locality.NeighborList.from_arrays(
                 4, 4, query_point_indices, point_indices, distances, weights)
 
+    def test_from_points(self):
+        box = freud.box.Box(2, 3, 4, 1, 0, 0)
+        testpoints = np.array([[0, 0, 0], [-2.2, -1.3, 2]])
+        testpoints_query = np.array(
+            [[-0.5, -1.3, 2.], [0.5, 0, 0], [-2.2, -1.3, 2.], [0, 0, 0.2]])
+        testpoint_indices = np.array([1, 0, 1, 0])
+        testquery_point_indices = np.array([0, 1, 2, 3])
```

Comment:
NeighborLists are always sorted by query point index. Do we enforce that the query point indices are sorted in the input, or do we perform that sorting? If perform that sorting on behalf of the user, I would like to be sure we test it by providing query point indices that are out of order. I can't recall.

Reply:
We enforce that query point indices are sorted in the input. I can add something to test for the error.

Reply:
Adding a test for out-of-order query point indices in `test_locality_NeighborList.py` would be wonderful!